.github/workflows/build_and_test_debug.yml

```diff
@@ -18,6 +18,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v2.3.4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
[Keeping your GitHub Actions and workflows secure Part 1](https://dev.to/petrsvihlik/using-environment-protection-rules-to-secure-secrets-when-building-external-forks-with-pullrequesttarget-hci) says "Combining pull_request_target workflow trigger with an explicit checkout of an untrusted PR is a dangerous practice that may lead to repository compromise."

Reply:
Do we really trust the article you shared?
I also don't quite get what the problem is. Is the PR test trying to use the secret in the user's repo? What's wrong with that?

Reply:
Right, so @jyyi1 wants to be able to develop off of his own forked repository. When making a PR from his forked repo, the secrets were withheld from his CI runs by GitHub for security reasons. As mentioned above, I was mistaken about how GitHub handles this. 

This means that anyone wanting to make a PR from their own fork will be unable, as their CI checks will fail due to the missing secrets, even if we approve those deployments. We already have the environment approvals set up, so we should be safe.

Reply:
Perhaps we do this for iOS only then, since the other workflows don't need secrets.

Reply:
Sure, I guess I can split this workflow up into two halves. We just can't have `pull_request` and `pull_request_target` events in the same workflow, otherwise it gets quite ugly - triggering each job twice on each push.